test/middleware_test.coffee

```diff
@@ -238,7 +296,7 @@ describe 'Middleware', ->
             {}
             middlewareFinished
             middlewareFailed
-          )
+          ).catch (reason) -> # supress warning re unhandled promise rejection
```

Comment:
Without that line 299, it was logging a warning in the tests, where middleware threw. Saying "unhandled promise rejection" - not a big deal, but simply catching the promise and doing nothing with it removes the warning. It would only effect people who's middleware is also throwing, in which case they'd see errors anyway.